flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkAppenderFactory.java

```diff
@@ -99,7 +122,8 @@ private RowType lazyPosDeleteFlinkSchema() {
 
   @Override
   public FileAppender<RowData> newAppender(OutputFile outputFile, FileFormat format) {
-    MetricsConfig metricsConfig = MetricsConfig.fromProperties(props);
+    MetricsConfig metricsConfig =
+        table != null ? MetricsConfig.forTable(table) : MetricsConfig.fromProperties(props);
```

Comment:
`MetricsConfig.fromProperties` is deprecated. I am wondering if we should ensure `table` is never null? That would require changing the current 2 constructors (technically breaking) without adding a new constructor. but this class should be an `@Internal` class. what do you think? @pvary @hililiwei @chenjunjiedada